server/hazard-service/src/main/java/edu/illinois/ncsa/incore/service/hazard/models/eq/utils/HazardCalc.java

```diff
@@ -115,6 +119,14 @@ public static SeismicHazardResult getGroundMotionAtSite(Earthquake earthquake, M
             demandUnits = BaseAttenuation.getUnits(hazardType);
         }
 
+        if (amplifyHazard && siteClassFC != null) {
+            SimpleFeature feature = GISUtil.getPointInPolygon(site.getLocation(), siteClassFC);
+            if (feature != null) {
+                String siteClass = feature.getAttribute(HazardUtil.SOILTYPE).toString();
```

Comment:
what should happen the the dataset doesn't have a "soiltype" attribute? Should the api call give an failure code for that point, or just continue execution to use a the default("D"?)?

If we want to do the latter option, then we should add a check here: 
`if(feature.getAttribute(HazardUtil.SOILTYPE) != null)`

Reply:
It should use the default site class and this is handled on line 307 of HazardCalc. If you look at that line, it checks if siteClass is null and if it is, it uses the model default.

Reply:
Added a null check before calling toString